src/XR/features/WebXRHandTracking.ts

```diff
@@ -89,6 +90,10 @@ export interface IWebXRHandTrackingOptions {
             right: string[];
             left: string[];
         };
+        /**
+         * The scene that contains the 3D UI elements. Passing this in turns on near interactions with the index finger tip
+         */
+        sceneForNearInteraction?: Scene;

```

Comment:
>sceneForNearInteraction [](start = 8, length = 23)

does the scene need to now be specified when initializing this feature? Is it possible to switch scenes while xr/hand tracking is still running?

Reply:
The scene has to be passed in, because nothing outside of the GUI3DManager has knowledge about the utilityLayer scene that UI is created on.

Right now, as far as I'm aware, the UI scene is overlaid on top of the main scene. So if the main scene changes, the utility layer scene (the one being passed in) will stay the same.

Reply:
if this object exists across multiple scenes, would we want logic for setting/removing the scene to check for collisions? Would someone ever want to specify multiple scenes containing buttons?

---
In reply to: [545398931](https://github.com/BabylonJS/Babylon.js/pull/9620#discussion_r545398931) [](ancestors = 545398931)

Reply:
I think that WebXRHandTracking doesn't work properly when switching scenes in general, since the hand mesh and joint meshes are all created on the current scene. It's something Raanan is going to look into, but the near interaction scene logic can be updated when the main hand scene logic is updated.